jenkinsherder.py

```diff
@@ -156,161 +209,170 @@ def add_line(title, value):
-
-    # Second, process the output. Note that this needs to be updated since
-    # treeherder-dev is dead.
-    return result_string, trsc, tjc
-
-    req = TreeherderRequest(
-        protocol='http',
-        host=config['treeherder']['repo']['host'],
-        project=config['treeherder']['repo']['project'],
-        oauth_key=config['treeherder']['credentials']['key'],
-        oauth_secret=config['treeherder']['credentials']['secret']
-    )
-
-    req.post(trsc)
-    req.post(tjc)
-    # maybe we should return all of this generated json in case somebody embeds this script?
+    job_details = get_result_summary(results)
+
+    # Populate jobs and submit job to treeherder (including log upload)
+    for j in [job1, job2]:
+        j.log_files += log_files
+        j.result = result_string
+        j.job_details += job_details
+        j.jenkins_build_tag = config['jenkins_build_tag']
+        j.jenkins_build_url = config['jenkins_build_url']
+        j.job_details.append({
+                    'url': j.jenkins_build_url,
+                    'value': 'Jenkins Build URL (VPN required)',
+                    'content_type': 'link',
+                    'title': 'artifact uploaded'})
+        j.job_details.append({
+                    'value': j.jenkins_build_tag,
+                    'content_type': 'text',
+                    'title': 'artifact uploaded'})
+        j.artifacts.append(('Results', 'json', results))
+        # TODO - fix this fake log parsing
+        if log_files:
+            failures = []
+            if results.get('total failed'):
+                failures = ['Total failed: %s' % results['total failed']]
+            j.parsed_logs[log_files[0]] = failures
+    if job1.build['repo'] == job2.build['repo']:
+        treeherder.submit_complete([job1, job2])
+    else:
+        # Jobs that belong to different repos cannot be submitted in one
+        # collection
+        treeherder.submit_complete([job1])
+        treeherder.submit_complete([job2])
+    return result_string
 
-    return result_string, trsc, tjc
 
 if __name__ == '__main__':
-    try:
-        result_string, trsc, tjc = main(sys.argv[1:])
-        print 'trsc = ' + json.dumps(json.loads(trsc.to_json()), sort_keys=True,
-                                     indent=4, separators=(',', ': '))
-
-        print 'tjc = ' + json.dumps(json.loads(tjc.to_json()), sort_keys=True,
-                                    indent=4, separators=(',', ': '))
-        if result_string == 'busted':
-            raise BaseException('Something went wrong in the test harness.')
-        elif result_string != 'success':
-            sys.exit(1)
-        else:
-            sys.exit(0)
-    except Exception as e:
-        print e
-        raise
-
+    result_string = main(sys.argv[1:])
+    if result_string == 'busted':
+        raise Exception('Something went wrong in the test harness.')
+    elif result_string != 'success':
+        sys.exit(1)
+    else:
+        sys.exit(0)
 
```

Comment:
Raising the exception without catching it anywhere will cause it to print and the script to exit with non-zero status. Here's an example: http://pf-jenkins.qa.mtv2.mozilla.com:8080/job/temp_treeherder_webrtc-nightly-win64-aurora-linux64/11/console